src/components/AdminRoute.js

```diff
@@ -1,12 +1,33 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Route, Redirect } from 'react-router-dom';
-import { auth } from '../firebase';
+import { db, auth } from '../firebase';
 import { useAuthState } from 'react-firebase-hooks/auth';
+import { useDocument } from 'react-firebase-hooks/firestore';
 
 export const AdminRoute = ({ component: Component, ...rest }) => {
   const { initialising, user } = useAuthState(auth);
-  if (initialising) {
-    return <Route {...rest} render={props => <div>Loading...</div>} />;
+  const [fetchedAccess, setFetchedAccess] = useState(false);
```

Comment:
The approach we're taking here is very procedural, and probably not necessary. I think it might be sufficient to do:

```suggestion
  const { loading, error, value } = useDocument(db.doc(`admins/${ user && user.email }`));
```

and when value is ready, we can act on it.

Reply:
That commit suggestion doesn't just apply to that line--if we just use `useDocument` all of the `useEffect` and fetch stuff goes away.

Reply:
Omg I hate resolve. And I can't do that because user might not exist until after useAuthState

Reply:
This will be resolve by this: https://github.com/CSFrequency/react-firebase-hooks/pull/10 hopefully soon. This repo is pretty small, so if we want to fork it or just copy and paste it into our repo at some point we can, and do what we want with the API.